deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/FileUploadCompletionNotification.java

```diff
@@ -0,0 +1,195 @@
+     */
+    public FileUploadCompletionNotification(String correlationId, Boolean isSuccess, Integer statusCode, String statusDescription)
+            throws IllegalArgumentException
+    {
+        setCorrelationId(correlationId);
+        updateStatus(isSuccess, statusCode, statusDescription);
+    }
+
+    /**
+     * Construct this notification with json
+     * @param json the json to parse.
+     */
+    public FileUploadCompletionNotification(String json)
+    {
+        Gson gson = new GsonBuilder().disableHtmlEscaping().serializeNulls().create();
+        FileUploadCompletionNotification fileUploadCompletionNotification;
+
+        try
+        {
+            fileUploadCompletionNotification = gson.fromJson(json, FileUploadCompletionNotification.class);
+        }
+        catch (Exception malformed)
+        {
+            throw new IllegalArgumentException("Malformed json:" + malformed);
+        }
+
+        this.correlationId = fileUploadCompletionNotification.getCorrelationId();
+        this.isSuccess = fileUploadCompletionNotification.getSuccess();
+        this.statusCode = fileUploadCompletionNotification.getStatusCode();
+        this.statusDescription = fileUploadCompletionNotification.getStatusDescription();
+    }
+
+    /**
+     * Update the status information in the collection, and return the new json.
+     *
+     * @param isSuccess is a Boolean representing whether the file was uploaded successfully.
+     * @param statusCode is the status for the upload of the file to storage.
+     * @param statusDescription is the description of the status code.
+     * @throws IllegalArgumentException if one of the parameters is null, empty, or not valid.
+     */
+    private void updateStatus(Boolean isSuccess, Integer statusCode, String statusDescription) throws IllegalArgumentException
+    {
+        this.isSuccess = isSuccess;
+        this.statusCode = statusCode;
+        this.statusDescription = statusDescription;
+    }
+
+    /**
+     * Convert this class in a valid json.
+     *
+     * @return a valid json that represents the content of this class.
+     */
+    public String toJson()
+    {
+        Gson gson = new GsonBuilder().disableHtmlEscaping().serializeNulls().create();
+
+        return gson.toJson(this);
+    }
+
+    /**
+     * Set the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
+     * for this upload from IoT Hub. Must equal {@link FileUploadSasUriResponse#getCorrelationId()}.
+     * @param correlationId the unique identifier that correlates this file upload status to a SAS URI that IoT Hub retreived from Azure Storage earlier.
+     */
+    public void setCorrelationId(String correlationId)
+    {
+        this.correlationId = correlationId;
+    }
+
+    /**
+     * @return Get the correlationId that correlates this FileUploadCompletionNotification to the earlier request to get the SAS URI
```

Comment:
Seems like this description describes the API and not the @return value. Maybe remove the 'Get'?